ktor-client/ktor-client-curl/desktop/test/io/ktor/client/engine/curl/test/CurlWebSocketTests.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.curl.test
+
+import io.ktor.client.*
+import io.ktor.client.engine.curl.*
+import io.ktor.client.plugins.websocket.*
+import io.ktor.websocket.*
+import kotlinx.coroutines.*
+import kotlin.test.*
+
+class CurlWebSocketTests {
+
+    private val TEST_WEBSOCKET_SERVER: String = "ws://127.0.0.1:8080"
+
+    @Test
+    fun testEcho() {
+        val client = HttpClient(Curl) {
+            install(WebSockets)
+        }
+
+        runBlocking {
+            client.webSocket("$TEST_WEBSOCKET_SERVER/websockets/echo") {
+                send(Frame.Text("Hello, world"))
+
+                val actual = incoming.receive()
+
+                assertTrue(actual is Frame.Text)
+                assertEquals("Hello, world", actual.readText())
+            }
+        }
+    }
+
+    @Test
+    fun testEmptyFrame() {
+        val client = HttpClient(Curl) {
+            install(WebSockets)
+        }
+
+        runBlocking {
+            client.webSocket("$TEST_WEBSOCKET_SERVER/websockets/echo") {
+                send(Frame.Text(""))
+
+                val actual = incoming.receive()
+
+                assertTrue(actual is Frame.Text)
+                assertEquals("", actual.readText())
+            }
+        }
+    }
```

Comment:
Would be good to add tests for other frame types, sending multiple frames.

Reply:
As I understand we have the following policy:
* Common tests are placed inside `:ktor-client:ktor-client-tests`
* Engine specific for corner cases as stored inside corresponding module.

It seems that the [`WebSocketTest`](https://github.com/ktorio/ktor/blob/a1d639cc38e7bd1becb3e27cc31451d55a4837ab/ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/WebSocketTest.kt#L25) covers the mentioned type of data ^.